app/plex_tools.py

```diff
@@ -138,24 +139,27 @@ def add_to_collection(config_path, plex, method, value, c, map, subfilters=None)
                     movies, missing = imdb_tools.tmdb_get_movies(config_path, plex, value, is_list=True)
                 elif method in ["tmdb_id", "tmdb_movie", "tmd_collection"]:
                     movies, missing = imdb_tools.tmdb_get_movies(config_path, plex, value)
-            if method == "trakt_list" and TraktClient.valid:
+            elif method == "trakt_list" and TraktClient.valid:
                 movies, missing = trakt_tools.trakt_get_movies(config_path, plex, value)
+            elif method == "tautulli" and Tautulli.valid:
+                movies, missing = imdb_tools.get_tautulli(config_path, plex, value)
     elif plex.library_type == "show":
         if (method in Show.__doc__ or hasattr(Show, method)):
             try:
                 shows = plex.Library.search(**{method: value})
             except PlexExceptions.BadRequest as e:
                 print(e)
         else:
-            if TraktClient.valid:
-                if method == "tmdb_list" and TMDB.valid:
-                    shows, missing = imdb_tools.tmdb_get_shows(config_path, plex, value, is_list=True)
-                elif method in ["tmdb_id", "tmdb_show"] and TMDB.valid:
-                    shows, missing = imdb_tools.tmdb_get_shows(config_path, plex, value)
-                elif method == "tvdb_show":
-                    shows, missing = imdb_tools.tvdb_get_shows(config_path, plex, value)
-                elif method == "trakt_list":
-                    shows, missing = trakt_tools.trakt_get_shows(config_path, plex, value)
+            if method == "tmdb_list" and TMDB.valid and TraktClient.valid:
```

Comment:
Does `TraktClient.valid` need to be True for `tmdb_list`?

Reply:
its odd but it does only for shows tho. right now the program uses trakt to covert a tmbd show id to its respective tvdb id. I don't like how it does this but I don't know of another way currently (Although I'm sure there is one). If you've got a better idea as to how to do this without Trakt please let me know

Reply:
we could probably add TVDB as a source using [this](https://github.com/spapanik/tvdb_api_client) but then its just one more api the user has to connect to

Reply:
Thanks for jogging my memory. I think we can leave it as is, considering Plex is working on a new TV agent.